### PR TITLE
Fix to confirm closing modal when changed data exists

### PR DESCRIPTION
**Pull request by WatanabeToshimitsu**
## What?
- Fix to confirm closing modal when changed data exists
- Refactoring

## Why?
- To improve UX
- To use react-hook-form

## See also
Related https://github.com/dataware-tools/dataware-tools/issues/104

## Screenshot or video
Confirm

### CODE REVIEW

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-user-manager/pull/244#" title="Author self-approved">WatanabeToshimitsu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-user-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-user-manager/blob/master/OWNERS)~~ [WatanabeToshimitsu]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by d-hayashi — APPROVED** [review]
LGTM!

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-user-manager/pull/244#pullrequestreview-852496107" title="Approved">d-hayashi</a>*, *<a href="https://github.com/dataware-tools/app-user-manager/pull/244#" title="Author self-approved">WatanabeToshimitsu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-user-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-user-manager/blob/master/OWNERS)~~ [WatanabeToshimitsu,d-hayashi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
